format/Flight.proto

```diff
@@ -525,3 +525,108 @@ message FlightData {
 message PutResult {
   bytes app_metadata = 1;
 }
+
+/*
+ * EXPERIMENTAL: Union of possible value types for a Session Option to be set to.
+ */
+message SessionOptionValue {
+  message StringListValue {
+    repeated string values = 1;
+  }
+
+  oneof option_value {
+    string string_value = 1;
+    bool bool_value = 2;
+    sfixed32 int32_value = 3;
+    sfixed64 int64_value = 4;
+    float float_value = 5;
+    double double_value = 6;
```

Comment:
Why all these alternatives? Why not simply `int_value` and `double_value`?

Reply:
Originally these types were chosen for consistency with APIs such as ODBC and JDBC. It'd be reasonable to prune these down to 64-bit integers and 64-bit doubles though.

Reply:
@pitrou @jduo you have more experience in this space than I do, can you come to a consensus perhaps along with @lidavidm and I'll do whatever is decided on.

Reply:
I wonder about the strong typing and how it would actually be used between a client and server without having both of them know in advance what the type is?

In the context of URL parsing, if we allow for passing extra options as query parameters, all of them would be untyped so how the client would be able to provide them to the server with the correct type?

Reply:
> I wonder about the strong typing and how it would actually be used between a client and server without having both of them know in advance what the type is?
> 
> In the context of URL parsing, if we allow for passing extra options as query parameters, all of them would be untyped so how the client would be able to provide them to the server with the correct type?

Very good point here.  I think the best behaviour is perhaps to have servers allow preferred-type-OR-string, and native Flight clients get the benefit of strongly-typed options, while other e.g. *DBC clients that have an interface that precludes passing in typed options (e.g. connection string parameters) can defer that to the server.  This should probably also be documented in the Protocol specification as the preferred practice.

Reply:
Per discussion here int32 and float will be removed, and recommendation to support string conversion fallback by server implementations will be added to documentation.